### ISSUE TITLE
0.8.1 won't build against ffmpeg 7.1

### ISSUE THREAD

**Issue by alucryd** [any1/neatvnc#121]
New ffmpeg release has moved a struct member: https://github.com/FFmpeg/FFmpeg/commit/7f17e0e6dd0b28b29295c1eeb1a38546eebc3946

Building 0.8.1 against the new ffmpeg results in the following error:
```
../neatvnc/src/h264-encoder.c: In function ‘h264_encoder_create’:
../neatvnc/src/h264-encoder.c:550:58: error: ‘AVFilterLink’ has no member named ‘hw_frames_ctx’
  550 |                 av_buffer_ref(self->filter_out->inputs[0]->hw_frames_ctx);
      |                                                          ^~
```

**Comment by any1** [comment]
Thanks for the heads-up.

We're abusing a "private" API for this, but there's really no other way to get at this. Maybe I can allocate my own `hw_frames_ctx` and put it there.

**Comment by any1** [comment]
Huh, looks like they may have added a function for it called `av_buffersink_get_hw_frames_ctx`.

**Comment by any1** [comment]
Should be fixed now; haven't built against 7.1 though.
